feature-libs/pickup-in-store/core/store/actions/pickup-location.action.ts

```diff
@@ -81,7 +82,30 @@ export const SetStoreDetailsSuccess = createAction(
   props<{ payload: PointOfService }>()
 );
 
-export const SetStoreDetailsFailure = createAction(
-  STORE_DETAILS_FAIL,
-  props<{ error: ErrorActionType }>()
-);
+interface TypeOf_SetStoreDetailsFailure {
+  (props: { error: ActionErrorProperty }): {
+    error: ActionErrorProperty;
+  } & TypedAction<typeof STORE_DETAILS_FAIL>;
+
+  /**
+   * @deprecated Use the `error` parameter with a non-null, non-undefined value.
+   *             Support for `null` or `undefined` will be removed in future versions,
+   *             along with the feature toggle `ssrStrictErrorHandlingForHttpAndNgrx`.
+   */
+  (props: { error: any }): { error: any } & TypedAction<
+    typeof STORE_DETAILS_FAIL
+  >;
+}
+
+// Note for future: when removing the old deprecated code, let's do the following:
+// - remove `TypeOf_SetStoreDetailsFailure`
+// - replace the implementation of `SetStoreDetailsFailure` with the following:
+// ```
+// export const SetStoreDetailsFailure = createAction(
+//   STORE_DETAILS_FAIL,
+//   props<{ error: ActionErrorProperty }>()
+// );
+// ```
+
+export const SetStoreDetailsFailure: TypeOf_SetStoreDetailsFailure =
+  createAction(STORE_DETAILS_FAIL, props<{ error: any }>());
```

Comment:
[NOTE]

It's a single exceptional case where NgRx actions are not classes (with constructors), but factories created with the `createAction` ngrx util.

This exceptional case required an exceptional solution.